ordered_product/views.py

```diff
@@ -0,0 +1,141 @@
+
+
+def client_check(user):
+    try:
+        return Client.objects.get(client_account=user)
+    except ObjectDoesNotExist:
+        return False
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def change_quantity(request, order_id):
+    try:
+        quantity = int(request.GET['quantity'])
+        ordered_product = OrderedProduct.objects.get(id=int(order_id))
+    except Exception:
+        return redirect('/client/')
+    if (quantity > 0):
+        ordered_product.increase_quantity(quantity)
+    elif (quantity < 0):
+        ordered_product.decrease_quantity(abs(quantity))
+    return redirect('/client/')
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def delete_order(request, order_id):
+    try:
+        OrderedProduct.objects.get(id=int(order_id)).delete_ordered_product()
+    except ObjectDoesNotExist:
+        pass
+    return redirect('/client/')
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def order(request):
+    return render(request, 'ordered_product/new_order.html', {})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def search_product_by_product_name(request):
+    prod_name = str(request.GET['product_name'])
+    message = ""
+    try:
+        product = Product.objects.get(product_name=prod_name)
+        supplier_products = SupplierProduct.objects.filter(qr_code=product)
+    except ObjectDoesNotExist:
+        supplier_products = []
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def get_products_by_client_area(request):
+    message = ""
+    client = client_check(request.user)
+    supplier_products = []
+    try:
+        deliveries_in_client_location = DeliveryLocation.objects.filter(location=client.area)
+        for delivery in deliveries_in_client_location:
+            supplier_products += list(SupplierProduct.objects.filter(user_name=delivery.user_name))
+    except Exception:
+        message = "No products found."
+    return render(request, 'ordered_product/new_order.html', {'supplier_products': supplier_products,
+                                                              'message': message})
+
+
+@user_passes_test(client_check, login_url='Main Page')
+def get_products_by_supplier_user_name(request):
+    supplier_user_name = str(request.GET['supplier_user_name'])
+    message = ""
+    supplier_has_delivery_in_client_area = False
+    try:
+        user = User.objects.get(username=supplier_user_name)
+        supplier = Supplier.objects.get(supplier_account=user)
+        deliveries = DeliveryLocation.objects.filter(user_name=supplier)
+        for delivery in deliveries:
+            if delivery.location == client_check(request.user).area:
+                supplier_has_delivery_in_client_area = True
```

Comment:
Why not use the `filter_by_supplier_and_location` method? 
```suggestion
        deliveries = DeliveryLocation.filter_by_supplier_and_location(supplier, client.area)
        if not deliveries:
            raise ObjectDoesNotExist
```